test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -1119,46 +1119,58 @@ class test_ucp_sockaddr_wireup_fail : public test_ucp_sockaddr_wireup {
             UCS_TEST_SKIP_R("cannot connect to server");
         }
 
-        if (fail_wireup_type == FAIL_WIREUP_MSG_SEND) {
+        while (!wait(e, wait_ep_flags) && (sender().get_err_num() == 0)) {
+            progress();
+        }
+
+        ucp_worker_h worker = e.worker();
+        switch (fail_wireup_type) {
+        case FAIL_WIREUP_MSG_SEND:
             /* Emulate failure of WIREUP MSG sending by setting the AM Bcopy
              * function which always return EP_TIMEOUT error */
-            UCS_ASYNC_BLOCK(&e.worker()->async);
+            UCS_ASYNC_BLOCK(&worker->async);
             for (ucp_lane_index_t lane_idx = 0;
                  lane_idx < ucp_ep_num_lanes(e.ep()); ++lane_idx) {
-                uct_ep_h uct_ep = e.ep()->uct_eps[lane_idx];
+                uct_ep_h uct_ep                = e.ep()->uct_eps[lane_idx];
                 uct_ep->iface->ops.ep_am_bcopy =
                         reinterpret_cast<uct_ep_am_bcopy_func_t>(
                                 ucs_empty_function_return_bc_ep_timeout);
             }
-            UCS_ASYNC_UNBLOCK(&e.worker()->async);
-        }
-
-        while (!wait(e, wait_ep_flags) && (sender().get_err_num() == 0)) {
-            progress();
-        }
+            UCS_ASYNC_UNBLOCK(&worker->async);
+            break;
+        case FAIL_WIREUP_MSG_ADDR_PACK:
+            if (!ucp_ep_config(e.ep())->p2p_lanes &&
+                (wait_ep_flags & UCP_EP_FLAG_SERVER_NOTIFY_CB)) {
+                /* Since no p2p transports selected on the endpoint, it sends
+                 * WIREUP_MSG/REPLY with empty addresses from the server */
+                UCS_TEST_SKIP_R("don't have p2p lanes to do address pack for");
+            }
 
-        if (fail_wireup_type == FAIL_WIREUP_MSG_ADDR_PACK) {
             /* Emulate failure of preparation of WIREUP MSG sending by setting
              * the device address getter to the function that always returns
              * error */
-            UCS_ASYNC_BLOCK(&e.worker()->async);
-            for (ucp_lane_index_t lane_idx = 0;
-                 lane_idx < ucp_ep_num_lanes(e.ep()); ++lane_idx) {
-                uct_ep_h uct_ep = e.ep()->uct_eps[lane_idx];
-                uct_ep->iface->ops.iface_get_device_address =
+            UCS_ASYNC_BLOCK(&worker->async);
+            for (ucp_rsc_index_t iface_id = 0; iface_id < worker->num_ifaces;
+                 ++iface_id) {
+                ucp_worker_iface_t *wiface                  =
+                        worker->ifaces[iface_id];
+                wiface->iface->ops.iface_get_device_address =
```

Comment:
minor: alignment is really hard to read, i'd better ignore alignment by `=`

Reply:
fixed